tests/unittest_scoped_nodes.py

```diff
@@ -1923,6 +1923,152 @@ def update(self):
         builder.parse(data)
 
 
+def test_issue940_metaclass_subclass_property():
+    node = builder.extract_node(
+        """
+    class BaseMeta(type):
+        @property
+        def __members__(cls):
+            return ['a', 'property']
+    class Parent(metaclass=BaseMeta):
+        pass
+    class Derived(Parent):
+        pass
+    Derived.__members__
+    """
+    )
+    inferred = next(node.infer())
+    assert isinstance(inferred, nodes.List)
+    assert [c.value for c in inferred.elts] == ["a", "property"]
+
+
+def test_issue940_property_grandchild():
+    node = builder.extract_node(
+        """
+    class Grandparent:
+        @property
+        def __members__(self):
+            return ['a', 'property']
+    class Parent(Grandparent):
+        pass
+    class Child(Parent):
+        pass
+    Child().__members__
+    """
+    )
+    inferred = next(node.infer())
+    assert isinstance(inferred, nodes.List)
+    assert [c.value for c in inferred.elts] == ["a", "property"]
+
+
+def test_issue940_metaclass_property():
+    node = builder.extract_node(
+        """
+    class BaseMeta(type):
+        @property
+        def __members__(cls):
+            return ['a', 'property']
+    class Parent(metaclass=BaseMeta):
+        pass
+    Parent.__members__
+    """
+    )
+    inferred = next(node.infer())
+    assert isinstance(inferred, nodes.List)
+    assert [c.value for c in inferred.elts] == ["a", "property"]
+
+
+def test_issue940_with_metaclass_class_context_property():
+    node = builder.extract_node(
+        """
+    class BaseMeta(type):
+        pass
+    class Parent(metaclass=BaseMeta):
+        @property
+        def __members__(cls):
```

Comment:
I'am probably nitpicking, but should not it be `def __members__(self)`?

Reply:
In the rest of the cases, the methods are on the metaclass and it's deliberately `cls` as a mnemonic to remind myself that they're supposed to take a class of <type> and not an instance of <class> (for which we'd use `self` by convention), e.g.
```python
class EnumMeta(type):
    def members(cls):
        print(f'{cls} of type {type(cls)}')

class Enum(metaclass=EnumMeta):
    pass

Enum.members()
# <class '__main__.Enum'> of type <class '__main__.EnumMeta'>
```

In this test, it's `cls` because I copied it from the test above and forgot to change it whoops.

Reply:
Makes sense. Thanks for taking the time to explain it!

Reply:
OK. Thanks for the explanation